codejam/client/client.py

```diff
@@ -1,315 +1,40 @@
-        parsed = Message(**json.loads(value))
-        callback = self.callbacks[cast(str, parsed.topic.type)][parsed.topic.operation]
-        callback(parsed)
-
-    def draw_line(self, message: Message) -> None:
-        """Draw lines from other clients"""
-        with self.canvas:
-            Color(hsv=message.value.data.colour)
-            line = Line(points=message.value.data.line, width=message.value.data.width)
-            self.ids[message.value.draw_id] = line
-
-    def draw_rectangle(self, message: Message) -> None:
-        """Draw rectangle from other clients"""
-        with self.canvas:
-            Color(hsv=message.value.data.colour)
-            rect = Rectangle(pos=message.value.data.pos, size=message.value.data.size)
-            self.ids[message.value.draw_id] = rect
-
-    @staticmethod
-    def display_error(message: Message) -> None:
-        """Display error modal."""
-        popup = ErrorPopup(
-            title=message.value.exception,
-            message=message.value.value,
-            error_code=message.value.error_id,
-        )
-        popup.open()
-
-
-class WhiteBoardScreen(Screen):
-    """WhiteBoardScreen"""
-
-    def on_pre_enter(self) -> None:
-        """Called when the screen is about to be shown."""
-        if not self.manager.ws:
-            self.manager.ws = asyncio.create_task(self.run_websocket())
-
-    async def run_websocket(self) -> None:
-        """Runs the websocket client and send messages."""
-        url = "ws://127.0.0.1:8000/ws/{0}".format(self.manager.username)
-        try:
-            print(url)
-            async with websockets.connect(url) as websocket:
-                try:
-                    while True:
-                        if m := self.wb.message:
-                            self.wb.message = ""
-                            print("sending " + m)
-                            await websocket.send(m)
-                        try:
-                            self.wb.received = await asyncio.wait_for(
-                                websocket.recv(), timeout=1 / 60
-                            )
-                        except asyncio.exceptions.TimeoutError:
-                            continue
-                        await asyncio.sleep(1 / 60)
-                except asyncio.CancelledError as e:
-                    print("Loop canceled", e)
-                finally:
-                    print("Loop finished")
-        except (ConnectionRefusedError, asyncio.exceptions.TimeoutError) as e:
-            print("Connection refused", e)
-
-
-class ErrorPopup(ModalView):
-    """ErrorPopup"""
-
-    title = StringProperty("")
-    message = StringProperty("")
-    error_code = StringProperty("")
-
-
-class Instructions(BoxLayout):
-    """Instructions rule"""
-
-    _canvas = ObjectProperty(None)
-
-
-class CanvasTools(BoxLayout):
-    """CanvasTools rule"""
-
-    ...
+from codejam.client.widgets.testcanvas import TestCanvas  # noqa: F401
```

Comment:
Can you also rename TestCanvas to DrawCanvas or whatever (something more in line with it's function)

Reply:
sure - there's likely many things that could use a rename